### ISSUE TITLE
⌨ Role menu purpose not apparent to screen readers

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#11973]
## ⌨ Accessibility Issue

The purpose of the new role dropdown menu lacks context for screen readers.

## 🦋 Expected Behaviour

The purpose of the menu is clear.

## 🕵️ Details

The new role dropdown menu uses the current role as the name which can be confusing. When using a screen reader, all you hear is something like "Button applicant collapsed, 3 items". For someone using the site who is not already aware of this, they may not understand what the dropdown does.

## 📋 Steps to Reproduce

1. Login with someone who has multiple roles `admin@test.com`
2. Start any screen reader
3. Tab focus until you land on the role dropdown
4. Observe what is read out

## 🙋‍♀️ Proposed Solution

~~We may want to add an `aria-label` and possibly an `aria-description` to notify users what the dropdown does 🤔 ~~

We took a look and unfortunately, `aria-description` is not fully supported yet. The interim solution would to add a label to the input like "Current role" and associate it with the button with an `aria-describedby` but that will require visual changes so we will need an updated design.

## Localized Copy

English: Select a different role to access its corresponding tools and information.
French: Sélectionnez un rôle différent pour accéder aux outils et informations correspondants.

## ✅ Acceptance Criteria

- [ ] Add "Current role" label (not a `label` element thought :sweat_smile: ) before the button (requires design change)
- [ ] Associated the current role specified text to the role menu as `aria-describedby`

**Comment by tristan-orourke** [comment]
@esizer and @NienkeBr, could you try to come up with a useful description for this?

**Comment by NienkeBr** [comment]
Would something like this work:
- Select a different role to access its corresponding tools and information.

**Comment by esizer** [comment]
After speaking with @MaddyDaigle , we think we may need to just stick some "Current role" prefix onto the button for now  while we explore better solutions.

Since this seems to be a novel pattern, we are still investigating how it can be immediately apparent to users on the purpose of the drop down.

**Comment by mnigh** [comment]
@esizer and @vd1992 are you still working on this issue?

**Comment by esizer** [comment]
> I don't see why this is still blocked, so removing the tag. I'm fine with Eric/Maddy suggestion for now.

It was blocked because it requries adding visual text to the navigation menu and wanted to give designers the opportunity to provide a design for the addition. If they are okay with just ading the prefix then it should be fine to unblock.

**Comment by gobyrne** [comment]
Ok, this is moving again on the design side. I'll close this older one as a duplicate.
